docs/framework/upgrade.md

```diff
@@ -0,0 +1,84 @@
+# Upgrade Reference
+
+!!! note "Scope"
+    This document discusses upgrade conventions used in CiviCRM core code. For extensions and modules which integrate with CiviCRM, upgrade conventions are different. See [Create a Module Extension](/extensions/civix.md) for more discussion of the module-focused upgrader.
+
+## Introduction
+
+Within CiviCRM Core code the upgrader classes handle making any necessary database changes either through PHP or SQL code, printing out any messages before or after the upgrade happens. Queuing up tasks for each version upgrade.
```

Comment:
I think its useful having the order of the upgrade steps so people are aware of how things do get run and when, Like the thing about verify DB state only happening on major upgrades is interesting